### PR TITLE
[APB-1524][PT]: Added Error Scenarios for when Client Responds and Er…

### PR DESCRIPTION
**Pull request by JTY009**
…ror Pages. Plus Updated Tests, Messages and Routes

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -78,9 +78,25 @@ client-complete.sub-header=What happens next
...
 
+# Form Errors
 error.confirmInvite.invalid=Please choose an option from below
 error.confirmTerms.invalid=Please accept the terms
 
+# Generic Issue Title
+client-problem.title=There is a problem
+
+# Not Signed Up
+not-signed-up.description=You are not signed up to <a href="www.google.com">report tax using software</a>.
```
Comment (Smudgey) [inline conf/messages]: Is this meant have a link to google embedded into the messages?

File: app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala [hunk 2]
```diff
@@ -41,7 +41,11 @@ class InvitationsService @Inject() (invitationsConnector: InvitationsConnector)
     } yield invitation
   }
 
-  def acceptInvitation(invitationId: String, mtdItId: MtdItId)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Int] = {
+  def acceptInvitation(mtdItId: MtdItId, invitationId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Int] = {
```
Comment (Smudgey) [inline app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala]: Was there a reason to swap these arguments :joy:

File: app/uk/gov/hmrc/agentinvitationsfrontend/connectors/InvitationsConnector.scala [hunk 3]
```diff
@@ -67,4 +70,12 @@ class InvitationsConnector @Inject() (
       http.PUT[Boolean, HttpResponse](acceptInvitationUrl(mtdItId, invitationId).toString, false).map(_.status)
     }
   }
+
+  def getClientInvitation(mtdItId: MtdItId, invitationId: String)(implicit hc: HeaderCarrier): Future[Option[Invitation]] = {
```
Comment (rvijayan) [inline app/uk/gov/hmrc/agentinvitationsfrontend/connectors/InvitationsConnector.scala]: You can use the ```getInvitation``` method. You just need to build the clientInvitationUrl separately.

File: conf/messages (hunk 1 above)
Comment (JTY009) [reply conf/messages]: That should not be there

File: conf/messages [hunk 4]
```diff
@@ -78,9 +78,25 @@ client-complete.sub-header=What happens next
...
+
+# Incorrect Invitation
+incorrect-invitation.description=The invitation you have accepted might be meant for someone else. Contact the tax agent who sent you the invitation.
+
+# Not Found Invitation
+not-found-invitation.description=The invitation you have accepted does not match our records. Contact the tax agent who sent you the invitation.
+
+# Invitation Already Responded
+invitation-already-responded.description=Someone else has already responded to the invitation you are trying to accept. Contact the tax agent who sent you the invitation.
```
Comment (Smudgey) [inline conf/messages]: As per discussion, can this be changes to: Someone has already responded to this invitation. If you think this is wrong, contact the tax agent who sent you the invitation.

**Pull request merged**